### PR TITLE
Update box.py

### PR DESCRIPTION
**Pull request by modanesh**
At the end of the discussion in this PR  #2374 , @zlig pushed a commit to fix a docstring but it was after the PR was merged. So it basically didn't make it into the master. As I looked into the modified files, `test_flatten_observation.py` and `test_spaces.py` were already updated. Also, as per our discussion with @jkterry1 ,  the `kellycoinflip` env has been removed from the gym. So, there's only this file `gym/spaces/box.py` left to be updated, which is what this PR does.

### CODE REVIEW

File: gym/spaces/box.py [hunk 1]
```diff
@@ -70,15 +71,17 @@ def _get_precision(dtype):
         high_precision = _get_precision(self.high.dtype)
         dtype_precision = _get_precision(self.dtype)
         if min(low_precision, high_precision) > dtype_precision:
-            logger.warn(f"Box bound precision lowered by casting to {self.dtype}")
```
Comment (ikamensh) [inline gym/spaces/box.py:73]: Not a useful change?

**Comment by ikamensh** [comment]
Looking at the changelog it's not obvious what is the intention of this change. Is it to use `._space` instead of `.space`? Why is it a good idea?

**Comment by modanesh** [comment]
Basically it's based on the pushed commit at the end of this discussion thread: #2374 . The main idea to fix a docstring.

**Comment by ikamensh** [comment]
I'm looking at the diff, and can't see a docstring changed, but instead a bunch of other changes.

**Comment by RedTachyon** [comment]
I'm confused, what's the point of this PR? It removes the seeding mechanism from init, which is in general a messy affair, but having the argument in the constructor is the least of its problems. It changes the logging to sometimes use `warnings` instead of always `gym.logger`, introduces an antiquated `str.format` instead of an existing f-string, and needlessly changes calls to `self.shape` to `self._shape`, even though the former is the getter for the latter.

I also don't see any docstring changes?

**Comment by jkterry1** [comment]
Thanks ariel, I assume this was a miscommunication on my end then

**Pull request closed without merging**